File: python/examples/list_jobs.py
"""Example: List and search for jobs in a Databricks workspace.

Prerequisites:
    - Set DATABRICKS_HOST and authentication env vars (token, OAuth, or Azure CLI)
    - pip install databricks-workspace-client
"""

from databricks_workspace_client import DatabricksWorkspaceClient

client = DatabricksWorkspaceClient()

# List all jobs (first 10)
print("=== First 10 jobs ===")
jobs = client.jobs.list_jobs(limit=10)
for job in jobs:
    print(f"  [{job.job_id}] {job.name} (created by: {job.creator})")

# Search by name
print("\n=== Search by name ===")
etl_jobs = client.jobs.list_jobs(name="etl")
for job in etl_jobs:
    print(f"  [{job.job_id}] {job.name}")

if not etl_jobs:
    print("  No jobs found matching 'etl'")

File: nodejs/examples/list-jobs.ts
/**
 * Example: List and search for jobs in a Databricks workspace.
 *
 * Prerequisites:
 *   - Set DATABRICKS_HOST and authentication env vars (token, OAuth, or Azure CLI)
 *   - Node.js >= 18
 */

import { DatabricksWorkspaceClient } from "../src/index.ts";

const client = new DatabricksWorkspaceClient();

// List first 10 jobs
console.log("=== First 10 jobs ===");
const jobs = await client.jobs.listJobs({ limit: 10 });
for (const job of jobs) {
  console.log(`  [${job.jobId}] ${job.name} (created by: ${job.creator})`);
}

// Search by name
console.log("\n=== Search by name ===");
const etlJobs = await client.jobs.listJobs({ name: "jb_test" });
for (const job of etlJobs) {
  console.log(`  [${job.jobId}] ${job.name}`);
}
if (etlJobs.length === 0) {
  console.log("  No jobs found matching 'jb_test'");
}
